src/modules/0.0.17/aws_secrets_manager/index.ts

```diff
@@ -152,24 +162,46 @@ class SecretMapper extends MapperBase<Secret> {
           'update'
         );
         if (isUpdate) {
-          if (secret.description !== cloudRecord.description) {
-            // we need to update the secret description
-            const input: UpdateSecretCommandInput = {
-              SecretId: secret.name,
-              Description: secret.description,
-            };
-            await this.updateSecret(client.secretsClient, input);
+          const input: UpdateSecretCommandInput = {
+            SecretId: secret.name,
+            Description: secret.description,
+          };
+          if (secret.value) {
+            input.SecretString=secret.value;
+          }
+          const updatedSecret = await this.updateSecret(client.secretsClient, input);
+          if (!updatedSecret) {
+            throw new Error("Secret not properly updated in AWS");
           }
 
-          if (secret.value !== cloudRecord.value) {
-            // we need to update the value
-            if (secret.value) {
-              const input: PutSecretValueCommandInput = {
-                SecretId: secret.name,
-                SecretString: secret.value,
-              };
-              await this.putSecretValue(client.secretsClient, input);
+          let finalSecret:Secret|undefined;
+          let i = 0;
+          do {
+            // retrieve updated secret to avoid race conditions
+            await new Promise(r => setTimeout(r, 2000)); // Sleep for 2s
+
+            const rawSecret = await this.getSecret(
+              client.secretsClient,
+              secret.name
+            );
+            finalSecret = await this.secretsMapper(rawSecret!);
+            i++;
+            if (!finalSecret) break;
```

Comment:
Do you really want `break` in these three, or `continue`?

Reply:
Actually i can set a continue if the value is undefined, and a break in the other 2 cases, as the version id is invalid

Reply:
So, I get the middle one being a `break`. It created the secret and has a new `versionId` that we want to use, so leaving at that point makes sense, but the last one I thought should be a `continue` as well, because somehow the updated secret doesn't have a value, and I don't understand what an update in that situation even means, and figured it was also some sort of transient error case that needs to be handled.

Reply:
Last one also contains the versionId that we want. In the case of not updating the value, the versionId does not change. So in this case, i just check that the return value contains a versionId and then i break the flow

Reply:
So it's the path taken when the `description` field is changed and nothing else is, I presume?

Reply:
Yes, that's it